core/src/main/java/org/apache/accumulo/core/clientImpl/ClientThreadPools.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.clientImpl;
+
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.accumulo.core.conf.ConfigurationCopy;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.util.threads.ThreadPools;
+
+public class ClientThreadPools {
+
+  public static class ThreadPoolConfig {
```

Comment:
I like this class being consolidated, for the most part, but it's not clear how having this consolidated here allows users to be able to have greater control over the thread pool lifecycles/error handling, which is what I thought was part of the goal. I don't see it exposed in the public API anywhere.

Reply:
Yeah, sorry for the confusion. I closed the prior pull request  #2340 where I was trying to allow the user to supply their own thread pools and uncaught exception handlers. It appeared based on the conversation in that PR that it was not going to be possible. This is a brand new PR which is only trying to do [this](https://github.com/apache/accumulo/pull/2346#issue-1044991496)